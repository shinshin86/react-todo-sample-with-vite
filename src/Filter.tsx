export type FilterType = "all" | "active" | "completed";

type Props = {
  filter: FilterType;
  setFilter: Function;
};

const Filter: React.FC<Props> = ({ filter, setFilter }) => {
  const onChange = (selectedFilter: string) => {
    setFilter(selectedFilter);
  };

  return (
    <div style={{ padding: 16 }}>
      <h2>Choose task filter</h2>
      <div>
        <input
          type="checkbox"
          id="all"
          name="all"
          value="all"
          checked={filter === "all"}
          onChange={(e) => onChange(e.target.value)}
        />
        <label htmlFor="all">All</label>
        <input
          type="checkbox"
          id="active"
          name="active"
          value="active"
          checked={filter === "active"}
          onChange={(e) => onChange(e.target.value)}
        />
        <label htmlFor="Active">Active</label>
        <input
          type="checkbox"
          id="completed"
          name="completed"
          value="completed"
          checked={filter === "completed"}
          onChange={(e) => onChange(e.target.value)}
        />
        <label htmlFor="completed">Completed</label>
      </div>
    </div>
  );
};

export default Filter;
