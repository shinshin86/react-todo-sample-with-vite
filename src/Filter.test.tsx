import { describe, expect, test } from "vitest";
import "@testing-library/jest-dom";
import Filter, { FilterType } from "./Filter";

import { render, screen } from "@testing-library/react";

describe("Simple working test", () => {
  const selectedFilter = "all";
  const emptySetFilter = (filter: FilterType) => console.log(filter);

  test("Should render correctly", () => {
    const { container } = render(
      <Filter filter={selectedFilter} setFilter={emptySetFilter} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("The text is visible", () => {
    render(<Filter filter={selectedFilter} setFilter={emptySetFilter} />);
    expect(screen.getByText("Choose task filter")).toBeInTheDocument();
  });

  test("Should checked all checkbox", () => {
    const { container } = render(
      <Filter filter={selectedFilter} setFilter={emptySetFilter} />,
    );
    expect(container.querySelector("#all")).toBeChecked();
  });

  test("Should active and complete are unchecked in the initial display", () => {
    const { container } = render(
      <Filter filter={selectedFilter} setFilter={emptySetFilter} />,
    );
    expect(container.querySelector("#active")).not.toBeChecked();
    expect(container.querySelector("#completed")).not.toBeChecked();
  });
});
