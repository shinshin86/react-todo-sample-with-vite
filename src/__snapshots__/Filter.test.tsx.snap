// Vitest Snapshot v1

exports[`Simple working test > Should render correctly 1`] = `
<div
  style="padding: 16px;"
>
  <h2>
    Choose task filter
  </h2>
  <div>
    <input
      checked=""
      defaultvalue="all"
      id="all"
      name="all"
      type="checkbox"
      value="all"
    />
    <label
      for="all"
    >
      All
    </label>
    <input
      defaultvalue="active"
      id="active"
      name="active"
      type="checkbox"
      value="active"
    />
    <label
      for="Active"
    >
      Active
    </label>
    <input
      defaultvalue="completed"
      id="completed"
      name="completed"
      type="checkbox"
      value="completed"
    />
    <label
      for="completed"
    >
      Completed
    </label>
  </div>
</div>
`;
